File: leon-core/skills/native/personal_agenda_skill/src/actions/create_event.ts
import type { ActionFunction } from '@sdk/types'
import { leon } from '@sdk/leon'

import { saveAgendaEvent } from '../lib/core-memory'

function formatDate(date: string): string {
  const parsed = new Date(`${date}T00:00:00Z`)
  return Number.isNaN(parsed.getTime())
    ? date
    : new Intl.DateTimeFormat('fr-FR', {
        day: '2-digit',
        month: '2-digit',
        year: 'numeric'
      }).format(parsed)
}

export const run: ActionFunction = async function (params) {
  const title = String(params.action_arguments['event_title'] || '').trim()
  const date = String(params.action_arguments['event_date'] || '').trim()
  const timeValue = params.action_arguments['event_time']
  const time = typeof timeValue === 'string' ? timeValue.trim() : ''

  if (!title || !date) {
    await leon.answer({ key: 'event_missing' })
    return
  }

  await saveAgendaEvent({ title, date, ...(time ? { time } : {}) })
  await leon.answer({
    key: 'event_created',
    data: {
      title,
      date: formatDate(date),
      time_suffix: time ? ` à ${time}` : ''
    }
  })
}

File: leon-core/skills/native/personal_agenda_skill/src/actions/create_event.py
from bridges.python.src.sdk.leon import leon
from bridges.python.src.sdk.types import ActionParams
from bridges.python.src.sdk.params_helper import ParamsHelper
from ..lib import memory


def run(params: ActionParams, params_helper: ParamsHelper) -> None:
    """Save a dated personal agenda event."""
    event_title = params_helper.get_action_argument('event_title').strip()
    event_date = params_helper.get_action_argument('event_date').strip()
    event_time = params.get('action_arguments', {}).get('event_time', '')
    time_suffix = f" à {event_time.strip()}" if isinstance(event_time, str) and event_time.strip() else ''

    event = {
        'title': event_title,
        'date': event_date,
        'time': event_time.strip() if isinstance(event_time, str) else ''
    }
    memory.add_event(event)

    leon.answer({
        'key': 'event_created',
        'data': {
            'title': event_title,
            'date': event_date,
            'time_suffix': time_suffix
        }
    })
